src/core/pyfunctions.cc

```diff
@@ -0,0 +1,1510 @@
+/*
+ *  OpenSCAD (www.openscad.org)
+ *  Copyright (C) 2009-2011 Clifford Wolf <clifford@clifford.at> and
+ *                          Marius Kintel <marius@kintel.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+
+#include <boost/format.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <Python.h>
+#include <pyopenscad.h>
+
+#include "primitives.h"
+#include "TransformNode.h"
+#include "RotateExtrudeNode.h"
+#include "LinearExtrudeNode.h"
+#include "CgalAdvNode.h"
+#include "CsgOpNode.h"
+#include "ColorNode.h"
+#include "RoofNode.h"
+#include "RenderNode.h"
+#include "SurfaceNode.h"
+#include "TextNode.h"
+#include "OffsetNode.h"
+#include "ProjectionNode.h"
+#include "ImportNode.h"
+
+#include "degree_trig.h"
+#include "printutils.h"
+#include "iofileutils.h"
+#include "handle_dep.h"
+
+//using namespace boost::assign; // bring 'operator+=()' into scope
+
+
+// Colors extracted from https://drafts.csswg.org/css-color/ on 2015-08-02
+// CSS Color Module Level 4 - Editor’s Draft, 29 May 2015
+extern  std::unordered_map<std::string, Color4f> webcolors;
+extern boost::optional<Color4f> parse_hex_color(const std::string& hex);
```

Comment:
I feel that parse_hex_color() could benefit from being made a utility function declared in a header file.
Now that webcolors is used outside of a single compilation unit, writing a lookup function instead of accessing the data structure directly would also be nice.

Reply:
PS! This could be done in a separate, "green refactoring" CL.
(I'm a huge fan of splitting out general cleanup into separate CL, so that reviewers can focus on actual functional changes)

Reply:
Better, read the color table from a text file.

Reply:
The list is coming from a fixed spec that's rarely updated. I don't see a huge benefit from putting it into a separate file.